### PR TITLE
ref(billing_metrics_consumer): Use commit policies

### PR DESCRIPTION
**Pull request by untitaker**
Arroyo has a new way to debounce commits. This may come with performance
regressions as we're still very much figuring things out, but we'll
invest more over time into making this perform.

Generally, standardizing on one way to do certain common things will
allow us to unify CLIs across consumers later and perhaps even move to
more declarative ways of creating consumers.

### CODE REVIEW

File: src/sentry/ingest/billing_metrics_consumer.py [hunk 1]
```diff
@@ -120,10 +107,12 @@ def close(self) -> None:
 
     def submit(self, message: Message[KafkaPayload]) -> None:
         assert not self.__closed
-        self.__messages_since_last_commit += 1
 
         payload = self._get_payload(message)
         self._produce_billing_outcomes(payload)
+        # TODO(markus): Perhaps call _bulk_commit directly here instead of
+        # manually accumulating offsets? Might be more in line with other
+        # consumers.
```
Comment (lynnagara) [inline src/sentry/ingest/billing_metrics_consumer.py:115]: I think you should call it directly and remove `_mark_commit_ready()` and `self.__ready_to_commit` dict as well.

**Comment by lynnagara** [comment]
You made a comment about performance regressions. A couple of questions about this -
1. Do you have an estimate of how much of a performance issue this is
2. If we switched this consumer over to the `ONCE_PER_SECOND` policy instead of counting messages (via the max batch size) - would we be able to avoid the performance regression?

**Comment by untitaker** [comment]
@lynnagara 

1. no, it is purely theoretical
2. yes, most likely

**Comment by untitaker** [comment]
I'll change it to the more common pattern, was considering to do that in a followup PR but it's probably fine

**Review by lynnagara — APPROVED** [review]

**Review by jjbayer — APPROVED** [review]

**Pull request merged**